File: Problem3.py
def triangle_display():
    print("Menu:")
    print("1. Display a right-angle triangle of ones")
    print("2. Display a Palindromic Triangle")
    print("3. Help")
    print("4. Exit")
def main():
    while True:
        triangle_display()
        choice = input("Enter your choice:")
        
        if choice == '1':
            def ones_triangle(lines):
                for i in range (1,lines+1):
                    print("1 "*i)
            lines = int(input("Enter the number of lines:"))   
            ones_triangle(lines)
        elif choice == '2':
            def Palindromic_triangle(rows):
                for i in range(1 , rows+ 1):
                    for j in range (1, i + 1):
                        print(j, end='')
                    for n in range (i-1, 0, -1):
                        print(n, end='')
                    print()
            rows =int(input("Enter the number of lines:"))      
            Palindromic_triangle(rows)
        elif choice == '3':
            print("""
Help:
A Palindromic Triangle is a triangular array of numbers where each row forms a Palindrome.
The first few lines of a Palindromic Triangle are:
1
11 
121 
12321 
1234321 
You can use pattern to draw a Palindromic Triangle for any number of lines.""")
        
        elif choice =='4':
            print("Exiting the Program.")
        else:
            print("Invalid choice!! Please choice a number from 1 - 4!")
main()

File: problem_3.cpp
#include <iostream>
using namespace std;

// Function to display the menu 
void menu() 
{ 
    cout << "1.Display a aright-angle traiangle of ones\n"; 
    cout << "2.Display a Palindromic traiangle \n"; 
    cout << "3.Help\n"; 
    cout << "4.Exit\n"; 
    
} 
// Function to display the result 
void result(int choice) 
{ 
 
    // Display the result 
    switch (choice) { 
    case 1: { 
        int i,j,rows;
    cout<<"Enter the number of lines : ";
        cin>>rows;
        for(i=1;i<=rows;i++)
        {
    for(j=1;j<=i;j++)
       cout<<"1";
       cout<<"\n";
          }
        break; 
    } 
    case 2: { 
        int rows;

    cout<<"Enter the row number: ";
    cin>>rows;
    for(int i=0;i<=rows;i++)
    {
       for(int j=1;j<=i;j++)
        cout<<j;
        for(int j=i-1;j>0;j--)
        cout<<j;
    cout<<"\n";
    }
        break; 
    } 
    case 3: { 
       cout<<"Help:\n A Palindromic traiangle is traiangle array of numbers where each row forms apalindrome.\n The first few lines of a Palindromic traiangle are:\n ";
        for(int i=0;i<=4;i++)
             {
            for(int j=1;j<=i;j++)
         cout<<j;
             for(int j=i-1;j>0;j--)
             cout<<j;
             cout<<"\n";
             }
              cout<<"You can use this pattren to draw a Palindromic traiangle for any number of lines. ";
        break; 
    } 
    case 4: { 
       cout<<"Exiting the program. ";
       exit(0);
        break; 
    } 
   
    default: 
     cout<<"Wrong Input\n";
      
    } 
} 

int main() {
      // Display the menu 
        menu(); 
     // Enter the choice 
        int choice; 
        cout << "Enter your choice:  "; 
         cin >> choice;

    // Display the result according to the choice 
        result(choice); 
 
    return 0;
   
    
}